openwrt/files/etc/udhcpc.user

```diff
@@ -1,19 +1,33 @@
 #!/bin/sh
 
+set -x
+
 case "$1" in
   # Same actions for renew or bound for the time being
   "renew"|"bound")
     # dump params to tmp so its easier to troubleshoot
     set > /tmp/dhcp.params
     radio0=`uci show 'wireless.radio0.channel'|cut -f 2 -d "'"`
     radio1=`uci show 'wireless.radio1.channel'|cut -f 2 -d "'"`
+
     if [ ! -z "$opt224" ] || [ ! -z "$opt225" ]; then
-      if [ "$opt224" != "$radio0" ] || [ "$opt225" != "$radio1" ]; then
```

Comment:
if the dhcp parameters haven't changed since the last time, it's a waste of time to set them, and issuing the commit can bounce the radios momentarily (and for the hostname, it does a reboot when changed)

This test detects that nothing changed, and bypasses any additional processing.

Reply:
suggest the following code

shopt -s nocasematch # set case insensitive match

    if [ ! -z "$opt224" ]  && [ ! -z "$opt225" ]; then
      if [ "$opt224" != "$radio0" ] || [ "$opt225" != "$radio1" ]; then
        if [ "$opt224" != "off" ]; then
          uci set 'wireless.radio0.channel'=$(printf %d "0x$opt224")
          uci set 'wireless.radio0.disabled' = 0
        else
          uci set 'wireless.radio0.disabled' = 1
        fi
        if [ "$opt225" != "off" ]; then
          uci set 'wireless.radio1.channel'=$(printf %d "0x$opt225")
          uci set 'wireless.radio1.disabled' = 0
        else
          uci set 'wireless.radio1.disabled' = 1
        fi
        uci commit
        wifi reload
      fi
    fi